airflow/settings.py

```diff
@@ -207,6 +209,35 @@ def configure_vars():
     DONOT_MODIFY_HANDLERS = conf.getboolean("logging", "donot_modify_handlers", fallback=False)
 
 
+def _run_openlineage_runtime_check():
+    """
+    Ensure compatibility of OpenLineage provider package and Airflow version.
+
+    Airflow 2.10.0 introduced some core changes (#39336) that made versions <= 1.8.0 of OpenLineage
+    provider incompatible with future Airflow versions (>= 2.10.0).
+    """
+    ol_package = "apache-airflow-providers-openlineage"
+    try:
+        ol_version = metadata.version(ol_package)
+    except metadata.PackageNotFoundError:
+        return
+
+    if (
+        ol_version
+        and Version(ol_version) < Version("1.8.0")
+        and Version(airflow_version) >= Version("2.10.0")
```

Comment:
We don't need to check for airflow_verion. By definition it's >=  2.10.0

Reply:
That's what i though, i figured i'd add it in case it gets cherry picked but yea, it may not make sense. Changing it.